### ISSUE TITLE
No messages from Android 

### ISSUE THREAD

**Issue by zenmanenergy** [phonegap/phonegap-plugin-push#85]
I'm not sure how to debug this problem.

I have gotten the plugin-push to work on IOS with my cordova app. I can get the registrationId, I can send a message from my php server to the apple APNS servers and the message shows up on my iphone 4s, but on the Android (v4.0) I have been only partially successful.

I have a server API key from the google developer console, I have gotten the registrationId to give me the token for the android. I have been able to send a message from the server to the GCM servers and it gives me a successful response.

But... nothing shows up on my android device, no error messages, no data when the app is in the foreground, no notifications in the background. I'm getting nothing.

I'm sure I did something wrong, but I'm not sure how to debug this at this point. Any suggestions?

Can someone post their android manifest file? I'd love to see if maybe I have messed up permission.

**Comment by macdonst** [comment]
You should have the following permissions in your AndroidManifest.xml

```
			<uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
			<uses-permission android:name="android.permission.WAKE_LOCK" />
			<uses-permission android:name="android.permission.VIBRATE"/>
			<uses-permission android:name="com.google.android.c2dm.permission.RECEIVE" />
			<permission android:name="$PACKAGE_NAME.permission.C2D_MESSAGE" android:protectionLevel="signature" />
			<uses-permission android:name="$PACKAGE_NAME.permission.C2D_MESSAGE" />
```

and these items in your application tag:

```
			<activity android:name="com.adobe.phonegap.push.PushHandlerActivity" android:exported="true"/>
			<receiver android:name="com.adobe.phonegap.push.CordovaGCMBroadcastReceiver" android:permission="com.google.android.c2dm.permission.SEND" >
				<intent-filter>
					<action android:name="com.google.android.c2dm.intent.RECEIVE" />
					<action android:name="com.google.android.c2dm.intent.REGISTRATION" />
					<category android:name="$PACKAGE_NAME" />
				</intent-filter>
			</receiver>
			<service android:name="com.adobe.phonegap.push.GCMIntentService" />
```

Although, I don't think that will be your problem if you are receiving a registrationId from GCM. If you look in example/server there is a quick Ruby script you can edit to send a message via GCM.

**Comment by behaghel** [comment]
@zenmanenergy can you share your code? and some logs?

**Comment by Nilos** [comment]
Did you use the correct sender id? If the sender id is incorrect, notifications are just not received.

**Comment by macdonst** [comment]
@zenmanenergy are you sending your push as data or as notification? See issue #87 which is now fixed in 1.2.1.

**Comment by macdonst** [comment]
@zenmanenergy please re-open if you still have the problem on 1.4.0 and can provide a reproduction scenario.
